.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+name: Kommons-Gated-PR-Tests
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: flanksource/build-tools:v0.13.6
+    steps:
+    - uses: actions/checkout@v2
+      with:
+      #Using Main branch of Karina Repository
+       repository: flanksource/karina
+    - run: |
+        module=$(cat go.mod | grep github.com/flanksource/kommons | awk '{print $2}')
+        go mod edit -replace github.com/flanksource/kommons@$module=github.com/${{ github.repository }}@${{ github.head_ref }}  
+        cat go.mod
+        go mod tidy
+      name: replace Module
+    - name: Test Make Linux
+      run: make linux
+    - uses: actions/upload-artifact@v2
+      with:
+        name: karina
+        path: ./.bin/karina

```

Comment:
You wouldn't need to upload a binary for this, 

for actual test execution this is the snippet that does the testing (we don't need a full matrix test though)
```
          GIT_API_KEY: ${{ secrets.GITHUB_TOKEN }}
          SUITE: ${{ matrix.suite }}
          KUBERNETES_VERSION: ${{matrix.k8s}}
          BUILD: test (${{matrix.k8s}}, ${{ matrix.suite }})
          ADDITIONAL_CONFIG: -c test/hosted-tests.yaml
        run: ./test/test.sh

```

Reply:
@moshloop Thanks for the Review, I decided to upload the artifacts because I initially thought of having the test in a separate Job like the Karina workflows, But I'll remove that and add the tests in the same job. Also, didn't add the actual tests as I was hoping to get a confirmation on the One's needed from the matrix ? Meanwhile I'll start with the baseline of v1.18.6, minimal.

Reply:
@moshloop  Added (v1.18.6,minimal) to the matrix and removed upload artifacts, In case we are needed to add more to the matrix I'd suggest going back to upload artifact and separate jobs for tests as to avoid building the binaries again and again.

Reply:
@moshloop Changed the logic to checkout the branch in question locally and using the local path for the module inside runner to prevent failures in the case of a fork merge where the repo name would change.